client/scripts/directives/previewPictures.js

```diff
@@ -0,0 +1,26 @@
+require('../app').directive('previewPictures', /* @ngInject */function () {
+  return {
+    scope: {
+      pictures: '=previewPictures'
+    },
+    require: 'previewPictures',
+    bindToController: true,
+    link: function ($scope, $element, $attrs, $ctrl) {
+      $element.on('click', $ctrl.preview)
+    },
+    controller: /* @ngInject */function ($attrs, $filter, Lightbox) {
+      var $ctrl = this
+      var authurl = $filter('authurl')
+
+      $ctrl.preview = function (event) {
+        event.preventDefault()
+        event.stopImmediatePropagation()
+
+        Lightbox.openModal($ctrl.pictures.map(function (picture) {
+          picture.url = authurl(picture.url)
```

Comment:
Can't remember if `authurl` support multiple application, but in any case it's better to create a new map with updated url